tests/test_route.py

```diff
@@ -623,3 +580,94 @@ def test_RouteAndNexthopInDifferentVrf(self, dvs, testlog):
         dvs.servers[2].runcmd("ip address del 20.0.0.2/24 dev eth0")
         dvs.servers[3].runcmd("ip route del default dev eth0")
         dvs.servers[3].runcmd("ip address del 20.0.1.2/24 dev eth0")
+
+class TestRoutePerf(TestRouteBase):
+    """ Performance tests for route """
+    def test_PerfAddRemoveRoute(self, dvs, testlog):
+        self.setup_db(dvs)
+        self.clear_srv_config(dvs)
+        numRoutes = 10000   # number of routes to add/remove
+
+        # generate ip prefixes of routes
+        prefixes = []
+        for i in range(numRoutes):
+            prefixes.append("%d.%d.%d.%d/%d" % (100 + int(i / 256 ** 2), int(i / 256), i % 256, 0, 24))
+
+        # create l3 interface
+        self.create_l3_intf("Ethernet0", "")
+        self.create_l3_intf("Ethernet4", "")
+
+        # set ip address
+        self.add_ip_address("Ethernet0", "10.0.0.0/31")
+        self.add_ip_address("Ethernet4", "10.0.0.2/31")
+
+        # bring up interface
+        self.set_admin_status("Ethernet0", "up")
+        self.set_admin_status("Ethernet4", "up")
+
+        # set ip address and default route
+        dvs.servers[0].runcmd("ip address add 10.0.0.1/31 dev eth0")
+        dvs.servers[0].runcmd("ip route add default via 10.0.0.0")
+
+        dvs.servers[1].runcmd("ip address add 10.0.0.3/31 dev eth0")
+        dvs.servers[1].runcmd("ip route add default via 10.0.0.2")
+
+        fieldValues = [{"nexthop": "10.0.0.1", "ifname": "Ethernet0"}, {"nexthop": "10.0.0.3", "ifname": "Ethernet4"}]
+
+        # get neighbor and arp entry
+        dvs.servers[0].runcmd("ping -c 1 10.0.0.3")
+
+        # get number of routes before adding new routes
+        startNumRoutes = len(self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"))
+
+        # add route entries
+        timeStart = time.time()
+        for i in range(numRoutes):
+            self.create_route_entry(prefixes[i], fieldValues[i % 2])
+
+        # wait until all routes are added into ASIC database
+        self.adb.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", startNumRoutes + numRoutes) # default timeout is 5 seconds
```

Comment:
We might have to wait and see how this goes in the PR tests. The PR runners can be a little slow so it is possible that we need to bump up the timeout a little bit for this test.

Reply:
Just kidding, it has run a few times successfully already it looks like. This default timeout is probably OK. :)